### PR TITLE
Update CI documentation following recent evolutions.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation

This allows checking the two first items at #6807.

Further improvements possible.

### CODE REVIEW

**Review by SkySkimmer — APPROVED** [review]

File: dev/ci/README.md [hunk 1]
```diff
@@ -54,7 +50,7 @@ Add a new `ci-mydev.sh` script to [`dev/ci`](/dev/ci) (have a look at
 set the corresponding variables in
 [`ci-basic-overlay.sh`](/dev/ci/ci-basic-overlay.sh); add the corresponding
 target to [`Makefile.ci`](/Makefile.ci); add new jobs to
-[`.travis.yml`](/.travis.yml) and [`.gitlab-ci.yml`](/.gitlab-ci.yml) so that
+[`.circleci/config.yml`](/.circleci/config.yml) and [`.gitlab-ci.yml`](/.gitlab-ci.yml) so that
```
[the remark this answers is not in the document]
Comment (Zimmi48) [reply dev/ci/README.md]: Yes, but this PR actually goes hand in hand with my request here: https://github.com/coq/coq/pull/7436#issuecomment-388346027
That's why I felt it was time to document how to run GitLab CI on private branches.

File: dev/ci/README.md [hunk 2]
```diff
@@ -126,7 +129,8 @@ Currently, available artifacts are:
 - the Coq executables and stdlib, in three copies varying in
   architecture and Ocaml version used to build Coq.
 - the Coq documentation, in two different copies varying in the OCaml
-  version used to build Coq
+  version used to build Coq. When submitting a documentation PR, this
```
[the remark this answers is not in the document]
Comment (Zimmi48) [reply dev/ci/README.md]: It was also my impression. Will fix this.

**Comment by Zimmi48** [comment]
Updated.

File: dev/ci/README.md [hunk 3]
```diff
@@ -36,10 +36,6 @@ On the condition that:
 - You do not push, to the branches that we test, commits that haven't been
   first tested to compile with the corresponding branch(es) of Coq.
 
-- Your development compiles in less than 35 minutes with just two threads.
-  If this is not the case, consider adding a "lite" target that compiles just
-  part of it.
-
 In case you forget to comply with these last three conditions, we would reach
```
Comment (ejgallego) [inline dev/ci/README.md]: I'd suggest we keep a paragraph about build time, even if we don't put a concrete number; "reasonable time" should be mentioned, there are a few Coq developments that may take in the order of > 10 hours.

File: dev/ci/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ejgallego) [reply dev/ci/README.md]: That's fine but I'd like to keep Travis a bit more. We can just wait to merge this PR tho.

File: dev/ci/README.md [hunk 4]
```diff
@@ -71,24 +67,38 @@ When you submit a pull request (PR) on Coq GitHub repository, this will
 automatically launch a battery of CI tests. The PR will not be integrated
 unless these tests pass.
 
-Currently, we have two CI platforms:
+We are currently running tests on many different platforms:
```
Comment (ejgallego) [inline dev/ci/README.md]: I feel that "many" is a bit overkill, why not just "We currently run tests on the following CI platforms:"

File: dev/ci/README.md [hunk 5]
```diff
@@ -71,24 +67,38 @@ When you submit a pull request (PR) on Coq GitHub repository, this will
...
+- GitLab CI is the main CI platform. It tests the compilation of Coq, of the
   documentation, and of CoqIDE on Linux with several versions of OCaml /
   camlp5, and with warnings as errors; it runs the test-suite and tests the
-  compilation of several external developments. It also tests the compilation
-  of Coq on OS X.
+  compilation of several external developments.
+
+- Circle CI runs tests that are redundant with GitLab CI and will be removed
+  eventually.
+
```
Comment (ejgallego) [inline dev/ci/README.md]: This discussion belongs elsewhere but there is still a good use case for Circle: chains of long depending contribs. There, Circle does a better job. So given that Circle maintenance overhead is quite low [using Docker] I wouldn't mind keeping some jobs like that on Circle.

File: dev/ci/README.md [hunk 6]
```diff
@@ -97,36 +107,29 @@ so that it doesn't, or provide a branch fixing these developments (or at
...
-
-GitLab specific information
----------------------------
+Advanced GitLab CI information
+------------------------------
 
-GitLab is set up to use the "build artifact" feature to avoid
+GitLab CI is set up to use the "build artifact" feature to avoid
 rebuilding Coq. In one job, Coq is built with `./configure -prefix
 install` and `make install` is run, then the `install` directory
```
Comment (ejgallego) [inline dev/ci/README.md]: AFAICT the prefix here is `_install_ci`.

**Review by ejgallego — COMMENTED** [review]
Looks largely OK to me, I would tho do another pass, in particular we have not yet decided what to do with Travis and Circle, [we can decide on Wed tho if Gaëtan is around], so I would either amend or wait.

File: dev/ci/README.md (hunk 5 above)
Comment (ejgallego) [reply dev/ci/README.md]: Well I guess it depends how do you read the sentence, maybe the current wording is OK.

**Comment by ejgallego** [comment]
Question on CODEOWNERS, shouldn't the rule be for `dev/ci/*` instead of only for the `.sh` part?

I didn't add an owner to Docker as I assumed the were covering the whole directory, but now I see this is not the case.

**Comment by SkySkimmer** [comment]
>shared runner status

What link is that?

Gaëtan Gilbert

**Comment by ejgallego** [comment]
> shared runner status
> 
> What link is that?

The one with the build graphs you posted the other day on Gitter

**Comment by ejgallego** [comment]
@Zimmi48 let me know if you want to address the comments, and to what extent; basically I am fine merging in the state you consider appropriate once you confirm you have read the suggestions.

File: dev/ci/README.md (hunk 3 above)
Comment (Zimmi48) [reply dev/ci/README.md]: Sounds good.

File: dev/ci/README.md (hunk 4 above)
Comment (Zimmi48) [reply dev/ci/README.md]: OK

File: dev/ci/README.md (hunk 6 above)
Comment (Zimmi48) [reply dev/ci/README.md]: Indeed.

**Comment by Zimmi48** [comment]
@ejgallego I've addressed your comments so that it can be merged now (including putting back the link to `.travis.yml`). We can update this document further when it is needed. Wrt the CODEOWNER thing, I will prepare an update.

**Comment by ejgallego** [comment]
Thanks, @SkySkimmer , could you re-post the gitlab status link here [my gitter search skills are blah]

**Comment by SkySkimmer** [comment]
https://monitor.gitlab.net/dashboard/db/ci

**Pull request merged**